devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy

```diff
@@ -0,0 +1,392 @@
+	log.info "SwitchBinaryReport called: ${endpoint}, ${cmd.value}"
+	def value = cmd.value ? "on" : "off"
+	endpoint ? changeSwitch(endpoint, value) : []
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd, endpoint = null) {
+	log.info "SecurityMessageEncapsulation called: ${cmd}, ${endpoint}"
+	def encapsulatedCommand = cmd.encapsulatedCommand(commandClassVersions)
+	if (encapsulatedCommand) {
+		waveEvent(encapsulatedCommand)
+	} else {
+		log.warn "Unable to extract encapsulatedCommand from $cmd"
+		createEvent(descriptionText: cmd.toString())
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd, endpoint = null) {
+	log.info "MultiChannelCmdEncap called: '${cmd}'  endpoint '${endpoint}'"
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	log.debug "MultiChannelCmdEncap: encapsulatedCommand '${encapsulatedCommand}'"
+	zwaveEvent(encapsulatedCommand, cmd.sourceEndPoint as Integer)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd, endpoint = null) {
+	log.info "NotificationReport called: notificationType: '${cmd.notificationType}', event: '${cmd.event}', endpoint '${endpoint}'"
+	def result = []
+
+	if (cmd.notificationType == 0x08) {
+		def value = cmd.event== 0x03? "on" : "off"
+		endpoint ? result = changeSwitch(endpoint, value) : []
+	} else if (cmd.notificationType == 0x04) {
+		def map = [name: "tamper"]
+		if (cmd.event == 0x02 || cmd.event == 0x04) {
+			map.value = detected
+		} else if (cmd.event == 0x06) {
+			map.value = clear
+		}
+		log.debug "NotificationReport eventmap: ${map}"
+		result << createEvent(map)
+		result
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.crc16encapv1.Crc16Encap cmd) {
+	log.info "Crc16Encap called: cmd '${cmd}'"
+	def versions = commandClassVersions
+	def version = versions[cmd.commandClass as Integer]
+	def ccObj = version ? zwave.commandClass(cmd.commandClass, version) : zwave.commandClass(cmd.commandClass)
+	def encapsulatedCommand = ccObj?.command(cmd.command)?.parse(cmd.data)
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	}
+	[:]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd) {
+	log.info "SensorMultilevelReport called, ${cmd}"
+	def map = [:]
+	def result = []
+	switch (cmd.sensorType) {
+		case 1:
+			map.name = "temperature"
+			def cmdScale = cmd.scale == 1 ? "F" : "C"
+			map.value = convertTemperatureIfNeeded(cmd.scaledSensorValue, cmdScale, cmd.precision)
+			map.unit = getTemperatureScale()
+			break		
+		case 5:
+			map.name = "humidity"
+			map.value = cmd.scaledSensorValue.toInteger()
+			map.unit = "%"
+			break
+		default:
+			map.descriptionText = cmd.toString()
+	}
+	log.debug "SensorMultilevelReport, ${map}, ${map.name}, ${map.value}, ${map.unit}"
+	result << createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.versionv1.VersionReport cmd) {
```

Comment:
This can be omitted. This is also now included in the `zwaveInfo` object

Reply:
Ok. Thanks. I will remove this part.